integrations/standalone/tests/pageobjects/Composite.ts

```diff
@@ -0,0 +1,53 @@
+import { Locator, Page } from '@playwright/test';
+import { CodeEditor } from './CodeEditor';
+import { ResponsibleSelect } from './ResponsibleSelect';
+import { Select } from './Select';
+import { Table } from './Table';
+
+export abstract class Composite {
+  readonly page: Page;
+  protected readonly label: string;
+
+  constructor(page: Page, label: string) {
+    this.page = page;
+    this.label = label;
+  }
+
+  protected abstract locator(): Locator;
+
+  input(label: string) {
+    return this.locator().getByLabel(label);
+  }
+
+  checkbox(label: string) {
+    return this.locator().getByLabel(label);
+  }
+
+  select(label: string) {
+    return new Select(this.page, this.locator(), label);
+  }
+
+  responsibleSelect(label: string) {
+    return new ResponsibleSelect(this.page, this.locator(), label);
+  }
+
+  macroInput(label: string) {
+    return new CodeEditor(this.page, this.locator(), true, label);
+  }
+
+  macroArea(label: string) {
+    return new CodeEditor(this.page, this.locator(), true, label);
+  }
+
+  scriptInput(label: string) {
+    return new CodeEditor(this.page, this.locator(), true, label);
+  }
+
+  scriptArea() {
+    return new CodeEditor(this.page, this.locator(), false, undefined);

```

Comment:
if you define the label as `label?: string` instead of `label: string | undefined`, you can remove undefined here.

Reply:
Even if this too maybe result in the same thing, they don't mean the same:
- label?: string -> means that this parameter is optional
- label: string | undefined -> means that this parameter needs to be a string or undefined